### PR TITLE
Update Android Gradle plugin to 4.2

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
- Update Android Gradle plugin to 4.2
- Add Android linter to linting workflow

Some notes: 4.2 compiles to Java 8 by default so we can remove those settings, SARIF output is disabled by default so it should be enabled. The `lint` task does not run `detekt` by default.

### CODE REVIEW

File: .github/workflows/sdk-lint.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-gradle-
       - name: Run detekt task
```
Comment (nielsvanvelzen) [inline .github/workflows/sdk-lint.yaml:30]:
```suggestion
      - name: Run detekt and lint tasks
```

Comment (nielsvanvelzen) [reply .github/workflows/sdk-lint.yaml]: fixed it, thank you kind reviewer

File: jellyfin-platform-android/build.gradle.kts [hunk 2]
```diff
@@ -16,9 +16,6 @@ android {
 
 	compileOptions {
 		isCoreLibraryDesugaringEnabled = true
-
-		sourceCompatibility = JavaVersion.VERSION_1_8
-		targetCompatibility = JavaVersion.VERSION_1_8
```
Comment (Maxr1998) [inline jellyfin-platform-android/build.gradle.kts:21]: Even if it's enabled by default, shouldn't those be kept anyway?

Comment (nielsvanvelzen) [reply jellyfin-platform-android/build.gradle.kts:21]: No because they default ti Java 8 now

Comment (Maxr1998) [reply jellyfin-platform-android/build.gradle.kts:21]: I know, but when thinking about it yesterday when I read it in the blogpost, I thought it'd still make sense to keep those for future proofing. Then again, it'll probably stay on 8 for quite a while.

**Review by Maxr1998 — APPROVED** [review]

**Pull request merged**